### PR TITLE
Feature/persist workspaces to db

### LINKED ISSUE #423: Write workspaces into sqlite database
Instead of storing workspaces in shared preferences, we should write them in database. This is important so that backup/restore works and in order to implement sync #154 properly to support tabs.

### PR DESCRIPTION
**Pull request by tuomas2**
Closes #423

### CODE REVIEW

File: app/src/main/java/net/bible/android/control/page/window/WindowRepository.kt [hunk 1]
```diff
@@ -18,25 +18,20 @@
...
 import net.bible.service.history.HistoryManager
-
-import org.apache.commons.lang3.StringUtils
-import org.json.JSONArray
-import org.json.JSONException
-import org.json.JSONObject
-
-import java.util.ArrayList
-
+import java.util.*
```
[the remark this answers is not in the document]
Comment (tuomas2) [reply app/src/main/java/net/bible/android/control/page/window/WindowRepository.kt]: I agree, not good. I don't know how it slipped there. My IDE settings even seem to be fine (should not happen automatically)

[the remark this answers is not in the document]
Comment (tuomas2) [reply app/src/main/java/net/bible/android/control/page/window/WindowRepository.kt]: Fixed.

File: app/src/main/java/net/bible/android/control/page/CurrentPageManager.kt [hunk 2]
```diff
@@ -237,22 +217,25 @@ open class CurrentPageManager @Inject constructor(
...
+            currentBible.entity,
+            currentCommentary.entity,
+            currentDictionary.pageEntity,
+            currentGeneralBook.pageEntity,
+            currentMap.pageEntity,
+            currentPage.bookCategory.getName()
+        )
+
+    fun restoreFrom(pageManagerEntity: WorkspaceEntities.PageManager?) {
+        if(pageManagerEntity == null) return
```
[the remark this answers is not in the document]
Comment (tuomas2) [reply app/src/main/java/net/bible/android/control/page/CurrentPageManager.kt]: Changed

File: app/src/main/java/net/bible/android/control/page/window/WindowRepository.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (timbze) [reply app/src/main/java/net/bible/android/control/page/window/WindowRepository.kt]: There actually were some more that had java.util.*:
HistoryItem.kt
IntentHistoryItem.kt
KeyHistoryItem.kt
AppDatabase.kt
WorkspaceEntities.kt

**Pull request merged**